H2TauTau/python/proto/analyzers/LHEWeightAnalyzer.py

```diff
@@ -0,0 +1,181 @@
+        self.alpha_s = TH1F('alpha_s', 'alpha_s', 2, 0, 2)
+
+
+        iterator = runsHandle.product().headers_begin()
+
+        # scan iterator so that it includes weight information
+        for it in range(100):
+            flag = False
+            for itt in iterator.base().lines():
+                print itt
+                if itt.find('weight id') != -1: flag = True
+
+            if flag == True: break
+            else: iterator.next()
+        else:
+            print 'Could not find any weight information !!'
+
+        self.index = []
+        self.pdfindex = []
+        self.alphaindex = []
+
+        idx = 0
+
+        for a in iterator.base().lines(): 
+            if a.find('weight id')==-1: continue
+            print a
+
+            # For Higgs signal
+#            if (a.find('muR')!=-1 and a.find('muF')!=-1 and a.find('muR=2 muF=0.5')==-1 and a.find('muR=0.5 muF=2')==-1):
+            # For LO Z->tautau
+            if(a.find('mur')!=-1 and a.find('muf')!=-1 and a.find('mur=2 muf=0.5')==-1 and a.find('mur=0.5 muf=2')==-1):
+
+                self.index.append(idx)
+
+
+# For Higgs signal 
+#            if a.find('PDF')!=-1:
+#                flag = False
+#                for _id in range(260000, 260101):
+#                    if a.find(str(_id))!=-1: flag = True
+#
+#                if flag:
+#                    self.pdfindex.append(idx)
+#                
+#            if a.find('PDF')!=-1 and a.find('265000')!=-1 or a.find('266000')!=-1: 
+#                self.alphaindex.append(idx)
+
+
+# For LO Z->tautau
+            if a.find('Member')!=-1:
+                flag = False
+                for _id in range(10, 111):
+                    if a.find('id=\"' + str(_id) + '\"')!=-1: flag = True
+
+                if flag:
+                    self.pdfindex.append(idx)
+                
+#            if a.find('pdf')!=-1 and a.find('292301')!=-1 or a.find('292302')!=-1: 
+
+            idx += 1
+
+        self.alphaindex.append(0)
+        self.alphaindex.append(1)
+
+
+
+
+        print 'scale variation index: ', self.index, 'len', len(self.index)
+        print 'pdf variation index: ', self.pdfindex, 'len', len(self.pdfindex)
+        print 'alpha variation index: ', self.alphaindex, 'len', len(self.alphaindex)
+
+
+
+
+
+    def process(self, event):
+        self.readCollections( event.input )
+
+#        import pdb; pdb.set_trace()
+        # if not MC, nothing to do
+        if not self.cfg_comp.isMC:
+            return True
+
+        # Add LHE weight info
+
+        event.LHE_originalWeight = 1.0
+
+        if self.mchandles['LHEweights'].isValid()==False: 
+            print 'LHE information not available !'
+            return True
+
+        event.LHE_originalWeight = self.mchandles['LHEweights'].product().originalXWGTUP()
+        event.scale_variation = []
+        event.pdf_variation = []
+        event.alpha_variation = []
+
+        for idx, ii in enumerate(self.index):
```

Comment:
Are you sure the code is correct?
It is at least misleading since you're looping over an index but calling the enumeration variable "idx" and the index variable "ii". This is also true for the subsequent enumerations